test/generic/Matrix-test.jl

```diff
@@ -334,6 +334,46 @@ end
    B = S([-t - 1 (-t) -R(1); -t^2 (-t) (-t); -R(-2) (-t - 2) (-t^2 - t - 1)])
 
    @test -A == B
+
+   # Exact ring
+   S = MatrixSpace(ZZ, rand(0:9), rand(0:9))
+   A = rand(S, -1000:1000)
+
+   @test iszero(A + (-A))
+   @test A == -(-A)
+   @test -A == S(-A.entries)
+
+   # Exact field
+   S = MatrixSpace(GF(7), rand(0:9), rand(0:9))
+   A = rand(S)
+
+   @test iszero(A + (-A))
+   @test A == -(-A)
+   @test -A == S(-A.entries)
+
+   # Inexact ring
+   S = MatrixSpace(RealField["t"][1], rand(0:9), rand(0:9))
+   A = rand(S, 0:200, -1000:1000)
+
+   @test iszero(A + (-A))
+   @test A == -(-A)
```

Comment:
Is it a good idea to use equality here rather than isapprox? I can just see this breaking on some architecture.

Reply:
I didn't expect `-(-x) == x` to fail on some platform for `x in RealField`... Isn't the sign just a bit in the structure, which would mean that flipping this bit twice cancels the effect?
If you are not sure about the failure, I would say we keep it like this until there are reports asserting a failure. If you know for certain it fails on some platform, then let's use `isapprox`.